ntbs-service/Pages/Notifications/Edit/ClinicalDetails.cshtml.cs

```diff
@@ -164,21 +165,39 @@ public ContentResult OnGetValidateClinicalDetailsProperty(string key, string val
             return ValidateModelProperty<ClinicalDetails>(key, value, shouldValidateFull);
         }
 
+        public ContentResult OnGetValidateClinicalDetailsListProperty(string key, IEnumerable<string> valueList, bool shouldValidateFull)
+        {
+            List<NotificationSite> notificationSites = new List<NotificationSite>();
+            foreach (string value in valueList)
+            {
+                notificationSites.Add(new NotificationSite {
+                    SiteId = (int) ((SiteId) Enum.Parse(typeof(SiteId), value))
```

Comment:
Does `(int) Enum.Parse(typeof(SiteId), "2")` seem clearer? I don't think the intermediate cast is gaining us anything (and it should be quite clear we're dealing with a 'SiteId` type already due to `typeof(SiteId)`?